src/auth/auth-manager.js

```diff
@@ -0,0 +1,149 @@
+const jwt = require('jsonwebtoken');
+const User = require('../users/user-models').User;
+const Session = require('./auth-models').Session;
+
+/**
+ * Callback for sending the response to the client.
+ *
+ * @callback loginResponse
+ * @param {Object} err - The error.
+ * @param {string} token - The JWT token.
+ */
+
+/**
+ * Given a valid email/password, generates and returns a JWT.
+ * @param {string} email - The user's email address.
+ * @param {string} password - The user's password.
+ * @param {loginResponse} next - The callback function to run after this function
+ *     finishes.
+ */
+const login = (email, password, next) => {
+  User.findOne({ email: email }, (err, user) => {
+    if (err || !user) {
```

Comment:
If there are no results to a query, ```err``` will be null (because no error actually occurred), and ```user``` will be null because there were no results. http://stackoverflow.com/questions/18214635/what-is-returned-from-mongoose-query-that-finds-no-matches